pythia/seo/tablematcher.py

```diff
@@ -0,0 +1,202 @@
+import warnings
+
+import numpy as np
+from sunpy.util import SunpyUserWarning
+
+__all__ = ['TableMatcher']
+
+
+class TableMatcher:
+    """
+    Table Matcher Object for finding corresponding rows across two distinct dataframes.
+    """
+    def __init__(self, match_type='cosine'):
+        """
+        Parameters
+        ----------
+        match_type : str, optional
+            The row matching algorithm, by default 'cosine'
+
+        Raises
+        ------
+        SunpyUserWarning
+            If unrecognized match type is passed.
+        """
+        self.match_type = match_type
+
+        if self.match_type is not 'cosine' and self.match_type != 'euclidean':
```

Comment:
`is not` and `!=` in the same comparisson! What about:
```suggestion
        if self.match_type not in ['cosine', 'euclidean']:
```